src/platform/Zephyr/BLEManagerImpl.cpp

```diff
@@ -240,15 +240,26 @@ void BLEManagerImpl::DriveBLEState()
         {
             mFlags.Clear(Flags::kAdvertisingRefreshNeeded);
             err = StartAdvertising();
-            SuccessOrExit(err);
+            if (err != CHIP_NO_ERROR)
+            {
+                // Return prematurely but keep the CHIPoBLE service mode enabled to allow advertising retries
```

Comment:
I think I don't understand. Is there any other place in the code that calls `StartAdvertising` (I can't find it)? If no, it looks like we unregister Gatt service in the `StartAdvertising` method in case of failure, but here we enable the service back. Is there something I'm missing?

Reply:
If we disable the service mode here (as it used to be done before my change) we will not be able to drive the BLE state to reach the `StartAdvertising()` in line 242 after the previous advertising request failed. So, this modification allows to do the actual restart. In `StartAdvertising()` we unregister the service but we do not disable the service mode (please take a look at the description of `kChipoBleGattServiceRegister flag`). Semantically we apply the `kAdvertisingEnabled/kAdvertising` scheme to the BLE service being enabled/registered (`kCHIPoBLEServiceMode_*abled/kChipoBleGattServiceRegister`). On the other hand maybe we may consider to stop using service mode concept at all, as it seems redundant to me, and retry to advertise regardless of it (but maybe I'm not aware of some edge cases).

Reply:
Discussed F2F, my comments are not relevant.